downstairs/src/lib.rs

```diff
@@ -1945,10 +1942,31 @@ impl Downstairs {
      * active Upstairs without causing jobs to be incorrectly sent to the
      * newly active Upstairs.
      */
-    async fn work_lock(
+    fn work_mut(
+        &mut self,
+        upstairs_connection: UpstairsConnection,
+    ) -> Result<&mut Work> {
+        self.check_upstairs_active(upstairs_connection)?;
+        let active_upstairs = self
```

Comment:
Yes.

(In fact, this function **extends** the lifetime of the `&mut self` because it returns a borrow, so no one can touch the `Downstairs` until the returned `&mut Work` is dropped)